src/shogun/machine/LinearMachine.cpp

```diff
@@ -120,3 +138,54 @@ void CLinearMachine::store_model_features()
 {
 }
 
+float64_t CLinearMachine::compute_bias()
+{
+	if (!m_labels)
+		SG_ERROR("No labels set\n")
+
+	if (!features)
+		SG_ERROR("No features set\n")
+
+	if (m_labels->get_num_labels() != features->get_num_vectors())
+		SG_ERROR("Number of training vectors does not match number of labels\n")
+
+	SGVector<float64_t> outputs = apply_get_outputs(features);
+
+	int32_t num_vec=features->get_num_vectors();
+
+	Map<VectorXd> eigen_outputs(outputs,num_vec);
+	Map<VectorXd> eigen_labels(((CRegressionLabels*)m_labels)->get_labels(),num_vec);
+
+	return (eigen_labels - eigen_outputs).mean() ;
+}
+
+
+bool CLinearMachine::train(CFeatures* data)
+{
+	/* not allowed to train on locked data */
+	if (m_data_locked)
+	{
+		SG_ERROR("%s::train data_lock() was called, only train_locked() is"
+				" possible. Call data_unlock if you want to call train()\n",
+				get_name());
+	}
+
+	if (train_require_labels())
+	{
+		if (m_labels == NULL)
+			SG_ERROR("%s@%p: No labels given", get_name(), this)
+
+		m_labels->ensure_valid(get_name());
+	}
+
+	bool result = train_machine(data);
+
+	/** optional compute bias*/
+	if(bias_flag)
+		set_bias(compute_bias());
+
+	if (m_store_model_features)
```

Comment:
linear models do not need to store model features. Only kernel machine do need to do that (I might have introduced this nonsense years ago ;) )